syntax = "proto3";

import "google/protobuf/empty.proto";

package MyPackage;

service WateringCan{
    rpc getState(google.protobuf.Empty) returns (WateringCanState) {}
    rpc setState(WateringCanState) returns (google.protobuf.Empty) {}
}

message WateringCanState{
    enum State{
        OFF = 0;
        ON = 1;
    }
    State state = 1;
}

//python3 -m grpc_tools.protoc -I./proto --python_out=. --grpc_python_out=. ./proto/watering_can.proto